middle_end/flambda2/to_cmm/to_cmm_env.ml

```diff
@@ -500,7 +537,7 @@ let split_complex_binding ~env ~res (binding : complex binding) =
   match binding.bound_expr with
   | Split _ -> res, Already_split
   | Splittable_prim { dbg; prim; args } ->
-    let new_bindings, new_cmm_args =
+    let new_bindings, new_cmm_args, free_vars_of_new_cmm_args =
       new_bindings_for_splitting binding.order args
     in
     let new_cmm_expr, res = rebuild_prim ~dbg ~env ~res prim new_cmm_args in
```

Comment:
There should be a comment explaining that we're going to use `free_vars_of_new_cmm_args` as the free names of `new_cmm_expr` and why it is correct.

Reply:
comment added